### ISSUE TITLE
element is empty violation in FormGroup when not using legend

### ISSUE THREAD

**Issue by jerhard2** [carbon-design-system/carbon#8614]
I am implementing FormGroup without legend `legendText=""`.  

The accessibility violation is Rule: 1.3.1 Info and Relationships `element <legend> is empty`.

Could legend be conditionally rendered similar to https://github.com/carbon-design-system/carbon/pull/7904?

## Environment

> Operating system

> Browser

> Automated testing tool and ruleset

> Assistive technology used to verify

## Detailed description

> What version of the Carbon Design System are you using?

> What did you expect to happen?

> What happened instead?

> What WCAG 2.1 checkpoint does the issue violate?

## Steps to reproduce the issue

1. Step one
2. Step two
3. Step three
4. etc.

> Please create a reduced test case in CodeSandbox
>
> - Style and vanilla JS:
>   https://codesandbox.io/s/github/carbon-design-system/carbon/tree/main/packages/components/examples/codesandbox
> - React:
>   https://codesandbox.io/s/github/carbon-design-system/carbon/tree/main/packages/react/examples/codesandbox

## Additional information

- Screenshots or code
- Notes

**Comment by tw15egan** [comment]
@jerhard2 do you have a CodeSandbox you could share to demonstrate what you are trying to achieve?

**Comment by joshblack** [comment]
Going to close out this due to inactivity. Feel free to comment with steps to reproduce and we'd be happy to help out!
